### DISCUSSION TITLE
Feature request: Rotating RPC providers

### DISCUSSION THREAD

**Discussion by crazyrabbitLTC** [ponder-sh/ponder#372] [Idea / Feature Request]
Pulling RPC data can be very expensive at any sort of scale. Would it be possible to provide an array of rpc providers and some limit of rpc calls or other logic to determine when we should rotate to another rpc provider?

The goal here would be to reduce cost of running ponder by spreading over as many rpc providers as possible.

**Comment by typedarray** [comment]
Hi! We [recently added support](https://github.com/0xOlias/ponder/releases/tag/%40ponder%2Fcore%400.0.92) for passing a custom Viem Transport in `ponder.config.ts`.

The Fallback transport seems like it would work here - what do you think?
